liquid-rust-typeck/src/lib.rs

```diff
@@ -76,7 +77,12 @@ pub fn check<'tcx>(
 
     let constraint = pure_cx.into_fixpoint(&mut fcx);
 
-    fcx.check(genv.tcx, def_id, body.mir.span, constraint)
+    let qualifiers = qualifiers
+        .iter()
+        .map(|q| -> ty::Qualifier { lowering::LoweringCtxt::lower_qualifer(q) })
```

Comment:
I didn't know this syntax was valid. Can you omit the `->`?

```suggestion
        .map(|q| ty::Qualifier { lowering::LoweringCtxt::lower_qualifer(q) })
```

Reply:
The whole `map` there goes away when we move to lowering in callbacks.